src/Sarif/GitHelper.cs

```diff
@@ -87,8 +87,8 @@ public string GetCurrentCommit(string repoPath)
         public string GetBlame(string filePath)
         {
             return GetSimpleGitCommandOutput(
-                Path.GetDirectoryName(filePath),
-                args: $"blame -f --porcelain {Path.GetFileName(filePath)}",
+                GetTopLevel(filePath),
```

Comment:
Note, I used `GetTopLevel()` and not `GetRepositoryRoot()` because `GetTopLevel()` is what `IsRepositoryRoot()` is comparing against, inside of `GetSimpleGitCommandOutput()`. 

I assume that `GetRepositoryRoot()`'s caching makes it more efficient, so it may be better to use that. 
I hope someone with more familiarity can weigh in.

Reply:
Thanks for raising this issue! We'd typically want a unit test for this sort of thing, is there a likely test to update/add in GitHelperTests.cs?

Reply:
So I'd considered that, but based on the fact that the original PR introducing support for Git Blame (#2358) notes that 

> The code changes have been manually tested, but the PR currently does not introduce unit tests for the feature. The problem lies with mocking appropriate external services, since the relevant code creates local objects to interact with these external services (such as GitHelper).

I don't see how I could.

Reply:
Yes. You ran into some technical debt from the original effort, agree with your assessment.